File: tests/fixtures/php/UserController.php
<?php
namespace App\Http\Controllers;

use Illuminate\Routing\Controller;

/**
 * Manages users.
 */
#[Route('/users')]
class UserController extends Controller
{
    /**
     * Show a user.
     */
    #[Route('/{id}', methods: ['GET'])]
    public function show(int $id): User
    {
        return User::find($id);
    }

    #[Route('/', methods: ['POST'])]
    public function store(): User
    {
        return new User();
    }
}

interface UserService
{
    public function find(int $id): ?User;
}

trait HasTimestamps
{
    public function touch(): void {}
}

enum Status
{
    case Active;
    case Inactive;
}

function helper(): string
{
    return "x";
}

File: tests/fixtures/csharp/UserController.cs
namespace MyApp.Api.Controllers;

using Microsoft.AspNetCore.Mvc;

/// <summary>Manages users via the public API.</summary>
[ApiController]
[Route("api/users")]
public class UserController : ControllerBase
{
    private readonly IUserService _service;

    public UserController(IUserService service)
    {
        _service = service;
    }

    /// <summary>Look up a user by id.</summary>
    [HttpGet("{id}")]
    public async Task<User> GetUser(int id)
    {
        return await _service.FindAsync(id);
    }

    [HttpPost]
    [Authorize]
    public IActionResult CreateUser([FromBody] User u) => Ok(u);
}

public record User(int Id, string Name);

public interface IUserService
{
    Task<User> FindAsync(int id);
}

public enum Status { Active, Inactive }
